openwisp_monitoring/check/classes/iperf.py

```diff
@@ -0,0 +1,205 @@
+        result_udp = self._get_iperf_result(res, exit_code, device, mode='UDP')
+
+        if store:
+            # Store iperf_result field 1 if any mode passes, store 0 when both fails
+            iperf_result = result_tcp['iperf_result'] | result_udp['iperf_result']
+            self.store_result(
+                {**result_tcp, **result_udp, 'iperf_result': iperf_result}
+            )
+        device_connection.disconnect()
+        return {**result_tcp, **result_udp, 'iperf_result': iperf_result}
+
+    def _get_device_connection(self, device):
+        """
+        Returns an active SSH DeviceConnection for a device.
+        """
+        openwrt_ssh = UPDATE_STRATEGIES[0][0]
+        device_connection = DeviceConnection.objects.filter(
+            device_id=device.id,
+            update_strategy=openwrt_ssh,
+            enabled=True,
+            is_working=True,
+        ).first()
+        return device_connection
+
+    def _get_iperf_servers(self, organization_id):
+        """
+        Get iperf test servers
+        """
+        org_servers = app_settings.IPERF_SERVERS.get(str(organization_id))
+        return org_servers
+
+    def _exec_command(self, dc, command):
+        """
+        Executes device command
+        """
+        return dc.connector_instance.exec_command(command, raise_unexpected_exit=False)
+
+    def _connect(self, dc):
+        """
+        Connects device returns its working status
+        """
+        return dc.connect()
+
+    def _get_iperf_result(self, res, exit_code, device, mode):
+        """
+        Get iperf test result
+        """
+
+        res_dict = json.loads(res)
+        if mode == 'TCP':
+            if exit_code != 0:
+                logger.warning(
+                    f'Iperf check failed for "{device}", {res_dict["error"]}'
+                )
+                return {
+                    'iperf_result': 0,
+                    'sent_bps_tcp': 0.0,
+                    'received_bps_tcp': 0.0,
+                    'sent_bytes_tcp': 0.0,
+                    'received_bytes_tcp': 0.0,
+                    'retransmits': 0,
+                }
+            else:
+                # Gbps = Gigabits per second
+                # GB = GigaBytes
+                # Todo : Remove below coversion once
+                # https://github.com/openwisp/openwisp-monitoring/pull/397 get merged
+                sent_json = res_dict['end']['sum_sent']
+                recv_json = res_dict['end']['sum_received']
+                sent_bytes = sent_json['bytes']
+                sent_bytes_GB = sent_bytes / 1000000000
+                sent_bps = sent_json['bits_per_second']
+                sent_Gbps = sent_bps / 1000000000
+                received_bytes = recv_json['bytes']
+                received_bytes_GB = received_bytes / 1000000000
+                received_bps = recv_json['bits_per_second']
+                received_Gbps = received_bps / 1000000000
```

Comment:
Didn't we decide to not do this conversion here? Please correct me if I am wrong.